src/com/google/javascript/jscomp/deps/BrowserModuleResolver.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2017 The Closure Compiler Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.javascript.jscomp.deps;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.javascript.jscomp.CheckLevel;
+import com.google.javascript.jscomp.ErrorHandler;
+import com.google.javascript.jscomp.JSError;
+import javax.annotation.Nullable;
+
+/**
+ * Resolution algorithm for Browsers.
+ *
+ * <p>Only unambiguous paths are supported. Paths must specify a file extension.
+ */
+public class BrowserModuleResolver extends ModuleResolver {
+
+  public BrowserModuleResolver(
+      ImmutableSet<String> modulePaths,
+      ImmutableList<String> moduleRootPaths,
+      ErrorHandler errorHandler) {
+    super(modulePaths, moduleRootPaths, errorHandler);
+  }
+
+  @Nullable
+  public String resolveJsModule(
+      String scriptAddress, String moduleAddress, String sourcename, int lineno, int colno) {
+
+    if (ModuleLoader.isAmbiguousIdentifier(moduleAddress)) {
+      if (errorHandler != null) {
```

Comment:
Could we forbid null for errorHandler and just make the creator pass in a no-op Errorhandler if they don't care about errors?